google-maps-workitem/src/main/java/org/jbpm/workitem/google/maps/GoogleMapsGeocodingWorkitem.java

```
@@ -0,0 +1,164 @@
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "SearchType", required = true),
+                @WidParameter(name = "SearchLocation", required = true),
+                @WidParameter(name = "Bounds"),
+                @WidParameter(name = "LocationType")
+        },
+        results = {
+                @WidResult(name = "GeocodingResults")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "google,maps,geocoding,locations",
+                action = @WidAction(title = "Get geocoding information using Google Maps")
+        ))
+public class GoogleMapsGeocodingWorkitem extends AbstractLogOrThrowWorkItemHandler {
+
+    private String apiKey;
+    private GeoApiContext geoApiContext;
+    private GoogleMapsAuth auth = new GoogleMapsAuth();
+    private static final String RESULTS_VALUE = "GeocodingResults";
+    private static final Logger logger = LoggerFactory.getLogger(GoogleMapsGeocodingWorkitem.class);
+
+    public GoogleMapsGeocodingWorkitem(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        String searchType = (String) workItem.getParameter("SearchType");
+        String searchLocation = (String) workItem.getParameter("SearchLocation");
+        String bounds = (String) workItem.getParameter("Bounds");
+        String locationType = (String) workItem.getParameter("LocationType");
+
+        try {
+
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+            if (geoApiContext == null) {
+                geoApiContext = auth.authorize(apiKey);
```

Why not getting this (authorize) in the constructor so it could fail fast?